src/psyclone/tests/psyir/transformations/profile_test.py

```diff
@@ -814,3 +816,56 @@ def test_omp_transform():
       CALL profile_psy_data%PostEnd'''
 
     assert correct in code
+
+
+def test_auto_invoke_return_last_stmt(parser):
+    ''' Check that using the auto-invoke profiling option avoids including
+    a return statement within the profiling region if it is the last statement
+    in the routine. '''
+    symbol_table = SymbolTable()
+    arg1 = symbol_table.new_symbol(
+        symbol_type=DataSymbol, datatype=REAL_TYPE)
+    zero = Literal("0.0", REAL_TYPE)
+    assign1 = Assignment.create(Reference(arg1), zero)
+    kschedule = KernelSchedule.create(
+        "work", symbol_table, [assign1, Return()])
+
+    Profiler.set_options([Profiler.INVOKES])
+    Profiler.add_profile_nodes(kschedule, Loop)
+    # The Return should be a sibling of the ProfileNode rather than a child
+    assert isinstance(kschedule[0], ProfileNode)
```

Comment:
One additional assert that theProfileNode contains the assignment? Again, extra careful admittedly - I wouldn't know what else it could contain :)

Reply:
Done.